context/app/static/js/shared-styles/charts/Histogram/Histogram.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useMemo } from 'react';
+import { Bar } from '@visx/shape';
+import { Group } from '@visx/group';
+import { AxisBottom, AxisLeft } from '@visx/axis';
+import { withParentSize } from '@visx/responsive';
+import { GridRows } from '@visx/grid';
+import { scaleLinear } from '@visx/scale';
+import { bin } from 'd3-array';
+
+function Histogram({ parentWidth, parentHeight, visxData, margin }) {
+  const xWidth = parentWidth - margin.left - margin.right;
+  const yHeight = parentHeight - margin.top - margin.bottom;
+
+  const binFunc = bin();
+  const chartData = binFunc(visxData);
+
+  const xScale = useMemo(
+    () =>
+      scaleLinear({
+        domain: [chartData[0].x0, chartData[chartData.length - 1].x1],
+        range: [0, xWidth],
+      }),
+    [chartData, xWidth],
+  );
+  const yScale = useMemo(
+    () =>
+      scaleLinear({
+        domain: [0, Math.max(...chartData.map((d) => d.length))],
+        nice: true,
+        round: true,
+        range: [yHeight, 0],
+      }),
+    [chartData, yHeight],
+  );
```

Comment:
Were you running into problems without `useMemo`? or is scaleLinear expensive? A little leery of adding complexity unless the benefits are clear... If it stays, maybe add a note above?